cmd/lk/utils.go

```diff
@@ -153,6 +156,32 @@ func wrapWith(wrap string) func(string) string {
 	}
 }
 
+func randomID() string {
+	const charset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+	seed := rand.NewSource(time.Now().UnixNano())
+	r := rand.New(seed)
+
+	result := make([]byte, 16)
+	for i := range result {
+		result[i] = charset[r.Intn(len(charset))]
+	}
+	return string(result)
+}
+
+func useTempPath(permanentPath string) (string, func() error) {
+	tempPath := path.Join(os.TempDir(), randomID())
+	relocate := func() error {
+		if err := os.Rename(tempPath, permanentPath); err != nil {
+			// NOTE: on macOS, `os.TempDir()` points to `/var/folders/...`.
+			// Because this directory is not automatically cleaned up, we need
+			// to remove it explicitly on a failure to relocate.
+			return os.RemoveAll(tempPath)
```

Comment:
Good call. I suppose it's not appropriate to just `defer relocate()` immediately after because the files may not have been created at all? Although be deferring we'll just swallow those errors anyway.